basis/compiler/tree/propagation/simple/simple.factor

```diff
@@ -62,6 +64,10 @@ ERROR: invalid-outputs #call infos ;
     [ "outputs" word-prop ] bi*
     with-datastack check-outputs ;
 
+! Force literal? to f to prevent non-specified inlining.
+: copy-output-infos ( #call word -- infos )
```

Comment:
Can you give an example of the inlining this is preventing?

Reply:
From `compiler.tree.propagation.tests`:
```factor
TUPLE: foo bar ;
...
: that-thing ( -- class ) foo ;

{ f } [ [ that-thing new ] { new } inlined? ] unit-test
```
will fail, returning `{ t }`, because `[ that-thing new ] optimized.` shows
```factor
[
    f
    { foo 1 1 tuple 236985587512 foo 266632840776318362 }
    <tuple-boa>
]

```
although `that-thing` is not declared inline.  The other unit test failures are along the lines of that.

Other strange things also happen, e.g.:
```
IN: scratchpad : test ( x -- y class ) 1 + foo ;
IN: scratchpad \ test optimized.
[ 1 + \ foo ]
IN: scratchpad [ test ] optimized.
[ test ]
IN: scratchpad [ test new ] optimized.
[
    test ( 11235067 11235068 -- 11235067 ) f
    { foo 1 1 tuple 236985587512 foo 266632840776318362 }
    <tuple-boa>
]
```
The word seems to work as intended though, so this might just be a limitation somewhere in the tree debugger.

Since the intention of this PR is "only" to provide inferred output types, not preventing this seemed to result in much more aggressive optimization than expected.

Reply:
Okay, have to look into this more.  I'm not sure if this is an issue or not...

Reply:
Wait, your example seems busted.

```factor
TUPLE: foo bar ;
...
: that-thing ( -- class ) foo ;
```

All that ``that-thing`` should do is push the ``foo`` symbol:

```factor
IN: scratchpad \ that-thing optimized.
[ \ foo ]
```

So, why is it making a ``foo`` instance now?

Reply:
Oh never-mind, you're doing ``that-thing new``... hmmm.